src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -629,29 +706,52 @@ public void notifyModificationNodeChanged(UUID studyUuid, UUID nodeUuid) {
         emitNodesChanged(studyUuid, List.of(nodeUuid));
     }
 
-    private void getParentReportUuidsAndNamesFromNode(NodeEntity nodeEntity, boolean nodeOnlyReport, List<Pair<UUID, String>> res) {
+    private void fillNodesInBuildOrder(NodeEntity nodeEntity, boolean nodeOnlyReport,
+        Map<UUID, Pair<UUID, String>> defNodeIdToReport,
+        List<Pair<UUID, String>> uuidsAndNames) {
+
         AbstractNode node = repositories.get(nodeEntity.getType()).getNode(nodeEntity.getIdNode());
-        res.add(0, Pair.of(getReportUuid(nodeEntity.getIdNode()), node.getName()));
-        if (node.getType() == NodeType.NETWORK_MODIFICATION && !nodeOnlyReport) {
-            getParentReportUuidsAndNamesFromNode(nodeEntity.getParentNode(), false, res);
+        if (nodeEntity.getType() != NodeType.NETWORK_MODIFICATION) {
+            uuidsAndNames.add(0, Pair.of(node.getReportUuid(), "Root"));
```

Comment:
Better use ROOT_NODE_NAME instead of "Root"

Reply:
Done